novobi_sales_b2b/models/sale_request.py

```diff
@@ -0,0 +1,166 @@
+from odoo import api, models, fields, _
+from odoo.exceptions import UserError
+import logging
+logger = logging.getLogger(__name__)
+
+
+class SalesRequest(models.Model):
+    _name = 'sale.request'
+    _order = 'date_request desc, id desc'
+    _inherit = ['mail.thread', 'mail.activity.mixin']
+    _description = "Sales Request"
+    _check_company_auto = True
+
+    name = fields.Char(string='Request Reference', required=True, index=True, states={
+                       'draft': [('readonly', False)]}, default=lambda self: _('New'))
+    date_request = fields.Datetime(
+        string='Request Date', required=True, readonly=True, index=True, default=fields.Datetime.now)
+    date_confirm = fields.Datetime(
+        string='Confirm Date', readonly=True, index=True)
+    company_id = fields.Many2one(
+        'res.company', 'Company', required=True, index=True, default=lambda self: self.env.company)
+    sale_user_id = fields.Many2one(
+        'res.users', string='Salesperson', index=True, default=lambda self: self.env.user,
+        domain=lambda self: [('groups_id', 'in', self.env.ref('sales_team.group_sale_salesman').id)])
+    pricelist_id = fields.Many2one(
+        'product.pricelist', string='Pricelist', check_company=True, required=True,
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]")
+    currency_id = fields.Many2one(related='pricelist_id.currency_id', depends=[
+                                  "pricelist_id"], store=True, ondelete="restrict")
+    partner_id = fields.Many2one(
+        'res.partner', string='Retailer',
+        required=True, change_default=True, index=True)
+    request_state = fields.Selection([
+        ('pending', 'Pending'),
+        ('approved', 'Approved'),
+        ('rejected', 'Rejected'),
+        ('cancelled', 'Cancelled')
+    ], string='Request Status', default='pending')
+    fulfillment_state = fields.Selection([
+        ('ready', 'Ready'),
+        ('waiting', 'Waiting'),
+        ('done', 'Done')
+    ], string='Fulfillment Status', default='ready')
+    tag_ids = fields.Many2many(
+        'crm.tag', string='Tags')
+    order_id = fields.Many2one(
+        'sale.order', string='Sale order', ondelete='cascade')
+    request_line = fields.One2many(
```

Comment:
for o2m or m2m fields, follow naming convention in Odoo guideline. request_line_ids